R/volcanoPlot.R

```diff
@@ -156,142 +150,204 @@ volcanoPlot <- function(contrastDF,
+                                                        '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
+                                                        '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1]);
+                                                    } else {
+                                                        t.showInfoSpan(e, o.display);
+                                                    };
+                                                }; }}")
+        }
 
-    if (!is.null(pthresholdLine)) {
-        volcanoPlot <- volcanoPlot +
-            geom_hline(yintercept = -log10(pthreshold), color = pthresholdLine,
-                       alpha = 0.5, size = refLineThickness)
-    }
+        # Optional Decorations
+        sizeBy <- NULL
+        sizeByShowLegend <- FALSE
+        decorations <- list()
 
-    if (!is.null(foldChangeLines)) {
-        volcanoPlot <- volcanoPlot +
-            geom_vline(xintercept = foldChangeLines, color = symbolFill["Increased"],
-                       alpha = 0.5, size = refLineThickness) +
-            geom_vline(xintercept = -foldChangeLines, color = symbolFill["Decreased"],
-                       alpha = 0.5, size = refLineThickness)
-    }
+        if (sizeByIntensity == TRUE) {
+            sizeBy <- "LogInt"
+            sizeByShowLegend <- TRUE
+        }
+        if (!is.null(pthresholdLine)) {
+            pthresholdLine <- paste(c("rgba(", paste(c(paste(col2rgb(pthresholdLine, alpha = FALSE), collapse = ","), 0.5), collapse = ","), ")"), collapse = "")
+            decorations <- list(line = list(list(color = pthresholdLine, width = refLineThickness, y = -log10(pthreshold))))
+        }
 
-    # Add geneSym labels to increased & decreased genes
-    if (!missing(geneSymLabels) & !missing(geneSymCol)) {
-        # Filter contrastDF to changed genes
-        idx <- contrastDF[[geneSymCol]] %in% geneSymLabels
-        contrastDFsubset <- contrastDF[idx,]
-        volcanoPlot <- volcanoPlot +
-            geom_text_repel(data = contrastDFsubset, aes_string(x = x, y = y, label = geneSymCol),
-                            show.legend = FALSE)
-    }
+        if (!is.null(foldChangeLines)) {
+            decorations <- list(line = append(decorations$line, list(list(color = symbolFill[which(groupNames == "Increased")],
+                                                                          width = refLineThickness,
+                                                                          x     = foldChangeLines),
+                                                                     list(color = symbolFill[which(groupNames == "Decreased")],
+                                                                          width = refLineThickness,
+                                                                          x     = -1*foldChangeLines)
+            )))
+        }
```

Comment:
Fix this indentation please (pull "line = append..." down below decorations and line it up from there.

Reply:
fixed